lib/signs_ui_web/plugs/api_auth.ex

```diff
@@ -0,0 +1,26 @@
+defmodule SignsUiWeb.Plugs.ApiAuth do
+  @moduledoc "Authenticates API requests using a set of keys defined in the environment."
+
+  import Plug.Conn
+  require Logger
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    with [key] <- get_req_header(conn, "x-api-key"),
+         user when not is_nil(user) <- Map.get(keys(), key) do
+      Logger.info("api_authorized user=#{user}")
```

Comment:
This will end up logging quite a bit more events. I think it's probably fine, but it would be more convenient if the user could instead be logged as a bit of metadata on the other lines we log. I think that's possible but I'm not entirely sure how. If that is relatively easy, can you try that instead? If not, this approach works.

Reply:
I did briefly search around but didn't find an easy way to do this short of taking full control over how the request logs are output, and then you have to reimplement the default behavior yourself.

Reply:
Ah well.

Reply:
@digitalcora `Logger.metadata` isn't what you're looking for?

Reply:
@paulswartz Plausibly! I didn't know about that until just now. Unfortunately it doesn't seem to work as-is; adding the metadata doesn't change anything about how Phoenix logs the request. I will look into it a little bit more tomorrow to see if I'm missing something.

Reply:
You also need to update the Logger config to include that atom. Check the API config for an example.

Reply:
Ah, thanks, that seems to work... well, it does come out a bit awkward:

```
17:55:41.634 [info] POST /messages
17:55:41.634 messages_api_user=test_user_2 [info] Sent 201 in 65µs
```

I'll refer to our other apps and see if we can't format that a little better.